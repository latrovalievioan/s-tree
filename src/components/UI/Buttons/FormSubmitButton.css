button[type=submit] {
  background: var(--text-color);
  color: var(--secondary-bg-color);
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;

  width: 100px;
  height: 40px;
  border-radius: 16px;

  font-weight: bold;
}

button[type=submit]:disabled {
  background: var(--text-color);
  color: var(--secondary-bg-color);
  cursor: initial;
}

button[type=submit]:disabled:hover,button[type=submit]:disabled:focus {
  background: var(--text-color);
  color: var(--secondary-bg-color);
  cursor: initial;
}

button[type=submit]:hover, button[type=submit]:focus {
  background: var(--accent-color);
  color: var(--secondary-bg-color);
}
