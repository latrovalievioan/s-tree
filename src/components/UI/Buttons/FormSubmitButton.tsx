import React from 'react';
import './FormSubmitButton.css';
import { Spinner } from '@/components/UI/Spinner';

type Props = {
  isPending: boolean;
  isDisabled?: boolean;
};
export const FormSubmitButton: React.FC<Props> = ({
  isPending,
  isDisabled,
}) => {
  return (
    <button
      data-testid="form-submit"
      type="submit"
      disabled={isPending || isDisabled}
    >
      {isPending ? (
        <Spinner skin="dark" width={24} height={24} border={2} />
      ) : (
        'Done'
      )}
    </button>
  );
};
