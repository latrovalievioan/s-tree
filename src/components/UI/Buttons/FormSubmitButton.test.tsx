import { render, screen, cleanup } from '@testing-library/react';
import { FormSubmitButton } from './FormSubmitButton';

describe(FormSubmitButton, () => {
  afterEach(() => {
    cleanup();
  });

  it('Should render the component', () => {
    render(<FormSubmitButton isPending={false} isDisabled />);
    const formSubmitButton = screen.getByTestId('form-submit');
    expect(formSubmitButton).toBeInTheDocument();
    expect(formSubmitButton).toBeVisible();
  });

  it('Should disable the button when disabled=true', () => {
    render(<FormSubmitButton isPending={false} isDisabled />);
    const formSubmitButton = screen.getByTestId('form-submit');
    expect(formSubmitButton).toBeDisabled();
  });

  it('Should be disabled when isPending=true', () => {
    render(<FormSubmitButton isPending />);
    const formSubmitButton = screen.getByTestId('form-submit');
    expect(formSubmitButton).toBeDisabled();
  });

  it('Should be enabled when neither isPending=true or isDisabled=true', () => {
    render(<FormSubmitButton isPending={false} />);
    const formSubmitButton = screen.getByTestId('form-submit');
    expect(formSubmitButton).toBeEnabled();
  });

  it('Should be contain "Done" when isPending=false', () => {
    render(<FormSubmitButton isPending={false} />);
    const formSubmitButton = screen.getByTestId('form-submit');
    expect(formSubmitButton).toHaveTextContent('Done');
  });

  it('Should display a spinner when isPending=true', () => {
    render(<FormSubmitButton isPending />);
    const formSubmitButton = screen.getByTestId('form-submit');
    const spinner = screen.getByTestId('loading-spinner');
    expect(formSubmitButton).toContainElement(spinner);
  });

  it('Matches snapshot', () => {
    render(<FormSubmitButton isPending={false} isDisabled />);
    const formSubmitButton = screen.getByTestId('form-submit');

    expect(formSubmitButton).toMatchSnapshot();
  });
});
