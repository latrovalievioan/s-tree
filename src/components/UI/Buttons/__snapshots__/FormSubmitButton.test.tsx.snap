// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormSubmitButton > Matches snapshot 1`] = `
<button
  data-testid="form-submit"
  disabled=""
  type="submit"
>
  Done
</button>
`;
